### ISSUE TITLE
Confirm Dialog : Triggering Confirm Dialog causes Accessibility check failure

### ISSUE THREAD

**Issue by tarwork** [primefaces/primefaces#7242]
**Describe the defect**
Using the current Primefaces Showcase 10.0.0 and checking it with the Site Improve accessibility checker chrome browser plugin along with viewing source exposes a reproducible accessibility error.

**Reproducer**
I am using the Primefaces Showcase

**Environment:**
PF Version: 10.0.0
JSF + version: Mojarra 2.2.x
Affected browsers: Chrome

**To Reproduce**
Steps:

1. go to page : https://www.primefaces.org/showcase/ui/overlay/confirmDialog.xhtml?jfwid=9ca23
2. view source for dialog content you will see some thing like this : `<div id="j_idt302:j_idt305" class="ui-confirm-dialog ui-dialog ui-widget ui-widget-content ui-corner-all ui-shadow ui-hidden-container" role="dialog" aria-describedby="j_idt302:j_idt305_content" aria-hidden="true" aria-modal="true" aria-labelledby="j_idt302:j_idt305_title" style="width: 350px; height: auto;">...</div>` Please Note : role="dialog"
3. click on confirm button
4. click "NO"
5. view source for the dialog content you will see something like this : `<div id="j_idt302:j_idt305" class="ui-confirm-dialog ui-dialog ui-widget ui-widget-content ui-corner-all ui-shadow ui-hidden-container" role="" aria-describedby="j_idt302:j_idt305_content" aria-hidden="true" aria-modal="false" aria-labelledby="j_idt302:j_idt305_title" style="width: 350px; height: auto; visibility: visible; left: 702.5px; top: 196.5px; z-index: 1003; display: none;" aria-live="off">...</div>` Please note : role="" 
6. This happen whether you choose No or Yes actually.
This condition causes Site Improve to flag this code with the following warning:
````
Invalid WAI-ARIA role, state or property
 4.1.2 Name, Role, Value

A WAI-ARIA element contains roles, states or properties that are not allowed on this type of element.

Check the W3C specification for Accessible Rich Internet Applications Suite (WAI-ARIA) 
to see what is supported for this type of element.
````
 

**Expected behavior**
It looks like some component of the code is clearing out the value of the attribute name "role" after the No or Yes button is clicked. Which therefore will cause this component to fail accessibility checks from Site Improve.

Please Note that adding role="dialog" manually in the source via the inspect UI and refreshing the Site Improve checker will clear the warning message. So that is pretty definitively the issue.

**Comment by melloware** [comment]
I will investigate this...

**Comment by tarwork** [comment]
further testing shows that this case also happens for the p:dialog as well when choosing the Modal option.

https://www.primefaces.org/showcase/ui/overlay/dialog.xhtml?jfwid=e0659

the same tracing of the html code in the inspect window will show the role attribute go from role="dialog" to role

perhaps they are sharing the same code. If not, I can create a new issue for this component.

**Comment by melloware** [comment]
I think it's all sharing the same code.  I will fix it.

**Comment by melloware** [comment]
OK thanks for reporting PR submitted.
